txnkv/transaction/2pc.go

```diff
@@ -871,6 +873,14 @@ func (tm *ttlManager) close() {
 	close(tm.ch)
 }
 
+func (tm *ttlManager) reset() {
+	if !atomic.CompareAndSwapUint32((*uint32)(&tm.state), uint32(stateRunning), uint32(stateUninitialized)) {
+		return
+	}
+	close(tm.ch)
+	tm.ch = make(chan struct{})
```

Comment:
I think it has the risk of race condition. Is it possible the `keepAlive` goroutine uses the new `ch`?

Reply:
Ah, yes... I change to copy the channel into the `keepAlive` goroutine instead of using the one in `ttlManager`.